src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramUniformFactory.cpp

```diff
@@ -216,6 +216,50 @@ class UScreenSpaceTriangleInfo : public UniformGroup
 	iUniform uScreenSpaceTriangle;
 };
 
+class URasterInfo : public UniformGroup {
+public:
+	URasterInfo(GLuint _program) {
+		LocateUniform(uVertexOffset);
+		LocateUniform(uTexCoordOffset);
+	}
+
+	void update(bool _force) override {
+		const bool isNativeRes = config.frameBufferEmulation.nativeResFactor == 1 && config.video.multisampling == 0;
```

Comment:
Why multisampling should be off? It is serious limitation for native-res mode.

Reply:
The idea of the vertex shift is to move the vertices so that the only sample of the fragments falls in the same relative position as in the N64. In multisampling mode, there are several samples across the fragment, so moving the vertices doesn't help. I am not sure if the texture coordinate shift should be applied in this case either, I guess that the last time I tested it gave better results.

If you remember, there was this issue with it in Excitebike and it is probably not the only issue.
![excitebike64-000](https://user-images.githubusercontent.com/12202580/111898586-4de2d100-8a27-11eb-8d41-d896b22c477f.png)